### PR TITLE
[TF] Implement TrainCallback on batch 

### PR DESCRIPTION
**Pull request by adtygan**
Modified [d2l.tensorflow](https://github.com/d2l-ai/d2l-en/blob/master/d2l/tensorflow.py)'s TrainCallback class to match plots in MXNet and PyTorch versions.

[link to issue](https://github.com/d2l-ai/d2l-en/issues/2019)

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: d2l/tensorflow.py [hunk 1]
```diff
@@ -1481,7 +1481,7 @@ def load_data_fashion_mnist(batch_size, resize=None):
 
 class TrainCallback(tf.keras.callbacks.Callback):
     """A callback to visiualize the training progress.
-
+
```
Comment (AnirudhDagar) [inline d2l/tensorflow.py:1484]:
```suggestion

```

**Review by AnirudhDagar — COMMENTED** [review]
Thanks @adtygan for the PR. @terrytangyuan can you please review the PR?

**Review by terrytangyuan — COMMENTED** [review]
Can you post the screenshot to show that they are consistent across frameworks?

**Comment by adtygan** [comment]
Hello @terrytangyuan, please find the screenshots below which were generated from the preview version available at [http://preview.d2l.ai/d2l-en/PR-2193/](http://preview.d2l.ai/d2l-en/PR-2193/).

Chapter 7.6. Convolutional Neural Networks (LeNet) training screenshots

Screenshot (123)
Screenshot (124)
Screenshot (125)

Chapter 8.2. Networks Using Blocks (VGG) training screenshots

Screenshot (127)
Screenshot (128)
Screenshot (129)

**Comment by AnirudhDagar** [comment]
@terrytangyuan @adtygan those plots actually don't take the changes into affect. The change is made only in the d2l lib which is not the right approach. We need to change the notebook as well, where class TrainCallback is saved. As you can verify from the CI logs, none of these notebooks were run and nothing has changed.I'll take a look later today and try to run all the concerning notebooks and fix the issue.

**Comment by astonzhang** [comment]
@adtygan Thanks. Could you send another PR to replace your github id with your name in our acknowledgement:
https://github.com/d2l-ai/d2l-en/blob/master/chapter_preface/index.md

Besides, as @AnirudhDagar pointed out, please edit the class definition in https://github.com/d2l-ai/d2l-en/blob/master/chapter_appendix-tools-for-deep-learning/utils.md and run `d2lbook build lib` (you may need to `pip install git+https://github.com/d2l-ai/d2l-book`) to sync the changes into d2l/tensorflow.py. For code consistency I've reverted changes to d2l/tensorflow.py only. Thanks.

**Pull request merged**